File: human_comparsion_cpp/src/main/java/com/netposa/poseidon/human/init/ThreadExceptionHandler.java
package com.netposa.poseidon.human.init;

import org.apache.log4j.Logger;

public class ThreadExceptionHandler implements Thread.UncaughtExceptionHandler {
    private Logger logger = Logger.getLogger(this.getClass());
    @Override
    public void uncaughtException(Thread t, Throwable e) {
        logger.error(String.format("thread => %d, error msg is => %s, %s", t.getId(), e.getMessage(), e));
        Thread newThread = new Thread(t, t.getName());
        newThread.setUncaughtExceptionHandler(this);
        newThread.start();
    }
}

File: human_comparsion/src/main/java/com/netposa/init/ThreadExceptionHandler.scala
package com.netposa.init

import org.slf4j.LoggerFactory


/**
  * 线程处理过程中的异常处理工具
  */
class ThreadExceptionHandler extends Thread.UncaughtExceptionHandler {
  val LOG = LoggerFactory.getLogger("ThreadExceptionHandler")
  override def uncaughtException(t: Thread, e: Throwable) {
    LOG.error("thread=>%d,error msg is=>%s".format(t.getId(), e.getMessage()), e);
    val newThread = new Thread(t, t.getName());
    newThread.setUncaughtExceptionHandler(this);
    newThread.start();
  }

}
